### PR TITLE
refactor(tests): Migrate to pytest

### PR DESCRIPTION
**Pull request by yasek**
Convert tests to pytest
Remove ddt and testtols dependencies

Closes falconry/falcon#844

### CODE REVIEW

**Comment by yasek** [comment]
Could you please check travis ci? I'm not able to reproduce this fail.

**Comment by yasek** [comment]
Rebased to master

File: falcon/testing/base.py [hunk 1]
```diff
@@ -1,123 +0,0 @@
...
-import falcon
-import falcon.request
-from falcon.testing.helpers import create_environ
-from falcon.testing.srmock import StartResponseMock
-
-
-# NOTE(kgriffs): Since this class is deprecated and we will be using it
-# less and less for Falcon's own tests, coverage may be reduced, hence
-# the pragma to ignore coverage errors from now on.
-class TestBase(unittest.TestCase):  # pragma nocover
```
Comment (jmvrbanac) [inline falcon/testing/base.py]: We have to keep this around for now. While it's deprecated, we can't remove this until 2.0.

File: falcon/testing/test_case.py [hunk 2]
```diff
@@ -1,109 +0,0 @@
...
-except ImportError:  # pragma: nocover
-    import unittest
-
-import falcon
-import falcon.request
-from falcon.testing.client import TestClient
-from falcon.testing.client import Result  # NOQA - hoist for backwards compat
-
-
-class TestCase(unittest.TestCase, TestClient):
```
Comment (yasek) [inline falcon/testing/test_case.py]: Fixed

File: falcon/testing/base.py (hunk 1 above)
Comment (yasek) [inline falcon/testing/base.py]: Fixed

**Comment by yasek** [comment]
Rebased

**Pull request merged**